### PR TITLE
Add CUDA 11.3 and fix broken cuTENSOR variations

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: docker.py [hunk 1]
```diff
@@ -71,7 +71,7 @@
     'nccl2.6-cuda100', 'nccl2.6-cuda101', 'nccl2.6-cuda102',
     'nccl2.7-cuda101', 'nccl2.7-cuda102', 'nccl2.7-cuda110', 'nccl2.7-cuda111',
     'nccl2.8-cuda112',
-    'nccl2.9-cuda102', 'nccl2.9-cuda110',
+    'nccl2.9-cuda102', 'nccl2.9-cuda110', 'nccl2.9-cuda113'
```
Comment (kmaehashi) [inline docker.py:74]:
```suggestion
    'nccl2.9-cuda102', 'nccl2.9-cuda110', 'nccl2.9-cuda113',
```

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
Blocked by CUDA driver update in Jenkins slave.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Review by asi1024 — APPROVED** [review]

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
https://github.com/cupy/cupy/pull/5264 got merged!
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
Hmm there are still some async mempool failures...
```
00:13:32 FAILED cupy_tests/cuda_tests/test_memory.py::TestAllocator_param_1_{mempool='MemoryAsyncPool'}::test_allocator_thread_local
00:13:32 FAILED cupy_tests/cuda_tests/test_memory.py::TestAllocator_param_1_{mempool='MemoryAsyncPool'}::test_set_allocator
00:13:32 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_free_bytes
00:13:32 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_total_bytes
00:13:32 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_total_bytes2
00:13:32 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_total_bytes_stream
00:13:32 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_used_bytes
00:13:32 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_used_bytes2
00:13:32 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_used_bytes_stream
```

**Comment by leofang** [comment]
I could try to take a look later tonight 😅

**Comment by leofang** [comment]
Hmmm I ran `pytest tests/cupy_tests/` on bare metal but couldn't reproduce the errors; all tests in `test_memory.py` passed. @kmaehashi is the CI hardware resource shared by more than one runners?

**Comment by kmaehashi** [comment]
@leofang Ah yes, several tests may run concurrently on the same device in Jenkins.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by leofang** [comment]
Let's see if https://github.com/cupy/cupy/pull/5308 fixes it...🤞

**Comment by leofang** [comment]
> Let's see if [cupy/cupy#5308](https://github.com/cupy/cupy/pull/5308) fixes it...🤞

Well, only part of it...
```
11:51:03 =========================== short test summary info ============================
11:51:03 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_free_bytes
11:51:03 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_total_bytes
11:51:03 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_total_bytes2
11:51:03 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_total_bytes_stream
11:51:03 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_used_bytes
11:51:03 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_used_bytes2
11:51:03 FAILED cupy_tests/cuda_tests/test_memory.py::TestMemoryAsyncPool::test_used_bytes_stream
11:51:03 = 7 failed, 80662 passed, 15834 skipped, 641 deselected, 74 xfailed, 1464 warnings in 5759.79s (1:35:59) =
```
Let me check later...

**Comment by leofang** [comment]
Should we mark these failing tests xfail for the time being? I do not immediately see what else could go wrong and I'm wondering if it's due to architecture difference.

**Comment by kmaehashi** [comment]
> Should we mark these failing tests xfail for the time being? I do not immediately see what else could go wrong and I'm wondering if it's due to architecture difference.

Agree, sent a PR. https://github.com/cupy/cupy/pull/5350

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
Jenkins test this please.

**Comment by kmaehashi** [comment]
Test failure is irrelevant.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**